boards/arm/gd32l233r_eval/gd32l233r_eval.dts

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2022 BrainCo Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/dts-v1/;
+
+#include <gigadevice/gd32l23x/gd32l233rc.dtsi>
+#include "gd32l233r_eval-pinctrl.dtsi"
+
+/ {
+	model = "GigaDevice GD32L233R Evaluation Kit";
+	compatible = "gd,gd32l233r-eval";
+
+	chosen {
+		zephyr,sram = &sram;
+		zephyr,flash = &flash0;
+		zephyr,console = &usart1;
+		zephyr,shell-uart = &usart1;
+		zephyr,flash-controller = &fmc;
+	};
+
+	soc {
+		/* SRAM0(16K) and SRAM1(16K) address is continuous. */
+		sram: memory@20000000 {
+			compatible = "mmio-sram";
+			reg = <0x20000000 DT_SIZE_K(32)>;
+		};
+	};
```

Comment:
question: why isn't this part of soc files?

Reply:
There doesn't have a real sram in here. But we introduce a trick usage at gd32l233r_eval board to get a larger sram.

Reply:
If SRAM0/1 has to be used as a single block, then let's define it this way in soc dts files.

Reply:
I`m fine with that if we keep SRAM0, SRAM1 and a combined region SRAM.

Reply:
done, move `sram` definition to `gd32l233rc` dts.